packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+       * Default value is false
+       */
+      shouldAllowTextStickersOnImages?: boolean;
+
+      /**
+       * Optional; indicate if the filter mode is enabled on the selected image
+       * Default value is false
+       */
+      enableFilterOnImages?: boolean;
+
+      /**
+       * Optional; Lets the developer specify the image output formats, more than one can be specified.
+       * Default value is Image.
+       */
+      imageFormatsAllowedToBeSelected?: ImageOutputFormats[];
+    }
+
+    /**
+     * Specifies the image output formats.
+     * @beta
+     */
+    export enum ImageOutputFormats {
+      /** Outputs image.  */
+      Image = 1,
+      /** Outputs pdf. */
+      PDF = 2,
+    }
+    const errorInvalidSource = 'Invalid input source.';
+    const errorInvalidCount = 'Invalid input count: Must supply the valid image counts (limit to 10).';
+    const errorInvalidResponse = 'Invalid response: Received more images than the specified max limit in the response.';
+    /**
+     * Capture one or multiple image(s) using camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than 10 or lesser than 1.
+     */
+    export async function captureImages(cameraImageInputs: CameraImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (
+        !cameraImageInputs ||
+        cameraImageInputs.maxVisualMediaCount > 10 ||
+        cameraImageInputs.maxVisualMediaCount < 1
+      ) {
+        throw errorInvalidCount;
+      }
+      if (cameraImageInputs.sourceProps.source != Source.Camera) {
+        throw errorInvalidSource;
+      }
+      // waiting the response from hub-SDK
+      const files = await sendAndHandleSdkError<VisualMediaFile[]>(
+        'visualMedia.image.captureImages',
+        cameraImageInputs,
+      );
+      // to ensure the number of images in the response is within the maximum limit.
+      if (files.length > cameraImageInputs.maxVisualMediaCount) {
+        throw errorInvalidResponse;
+      }
+      return files;
+    }
+
+    /**
+     * Upload the existing image(s) from the gallery.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than 10 or lesser than 1.
+     */
+    export async function retrieveImages(galleryImageInputs: GalleryImageProperties): Promise<VisualMediaFile[]> {
```

Comment:
Can we have something more specific name here? like `retrieveImagesFromGallery` as it'll be ambiguous as the code base scales.

Reply:
That's a great suggestion, and @TrevorJoelHarris also had the same suggestion, and we've talked about the naming last week. From the host app's perspective, calling this function won't block the host app user from selecting an image from the camera. The input GalleryImageProperties/CameraImageProp will inform the host app which visualMedia-related library they will be calling. For example, calling retrieveImages function will open the gallery in the host app, but user can still select image from camera from the UI on the gallery. `FromGallery` may bring confusion. But I'm open mind of the function naming. Would like to hear your thoughts.